dandischema/conf.py

```diff
@@ -0,0 +1,28 @@
+# This file defines the configuration for the DANDI schema
+
+from __future__ import annotations
+
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+
+class Config(BaseSettings):
+    """
+    Configuration for the DANDI schema
+
+    Note
+    ----
+        Since this class is subclass of `pydantic.BaseSettings`, each field of an
+        instance of this class can be populated from an environment variable of the
+        same name prefixed with the prefix defined in `model_config` with the name
+        of the environment variable interpreted case-insensitively.
+        For details, see https://docs.pydantic.dev/latest/concepts/pydantic_settings/
+    """
+
+    model_config = SettingsConfigDict(env_prefix="dandi_schema_")
+
+    nskey: str = "dandi"
+    id_prefix_pattern: str = r"[A-Z]+"
```

Comment:
only 1 to add: for `DATACITE_DOI_ID` (see below) and I think we would have minimal set needed. I might even remove `nskey` from being configurable here (although in the code still use nskey variable).

Reply:
and let's shorten to `id_pattern`

Reply:
done. `Config.id_prefix_pattern` is renamed to `Config.id_pattern, and ``Config.nskey` is drop since `"dandi"` is the namespace of DANDI ontology and doesn't change.